test/IDE/complete_ambiguous.swift

```diff
@@ -311,6 +313,31 @@ func testMissingArgs() {
 
   test3(after: Test.#^OVERLOADEDFUNC_MISSINGARG_BEFORE^#);
   test4(both: Test.#^OVERLOADEDFUNC_MISSINGARG_BEFOREANDAFTER^#)
+
+  enum Baz { case baz }
+
+  func trailing(x: Int, _ y: () -> Foo, z: () -> ())
+  func trailing(x: Int, _ y: () -> Bar, z: (() -> ())?)
+  func trailing(x: Int, _ y: () -> Baz, z: Int)
```

Comment:
Please add a couple more tests for `Any` and generic parameters to make sure they are handled properly as well.